docs/cdoc/index.rst

```diff
@@ -0,0 +1,14 @@
+===========================
+Qiskit C API (``qiskit.h``)
+===========================
+
+Quantum information:
+
+.. toctree::
+   :maxdepth: 1
+
+   qk-obs
+   qk-obs-term
+   qk-bit-term
+   exit-codes
```

Comment:
We might want to split this out into a separate toctree because it will be used by more than the sparse observable (see #14006). But we can do that later when we update the docs in a follow up.